pkg/cmd/cli/cmd/cancelbuild.go

```diff
@@ -105,7 +105,7 @@ func RunCancelBuild(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, arg
 	// Create a new build with the same configuration.
 	if cmdutil.GetFlagBool(cmd, "restart") {
 		request := &buildapi.BuildRequest{
-			ObjectMeta: kapi.ObjectMeta{Name: build.Name},
+			ObjectMeta: kapi.ObjectMeta{Name: build.Name, Namespace: namespace},
 		}
 		newBuild, err := client.Builds(namespace).Clone(request)
```

Comment:
Client-side substitution in resourcebuilder: https://github.com/openshift/origin/blob/master/Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource/visitor.go#L527